### PR TITLE
Add Rekt1800 keyboard

### PR DESCRIPTION
**Pull request by awkannan**
Add the REKT 1800 keyboard PCB to QMK.

The most important files to look at are config.h, rekt1800.h, and the keymaps.

The rest are boilerplate.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/rekt1800/info.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (awkannan) [inline keyboards/cannonkeys/rekt1800/info.json]: TODO: Update this

File: keyboards/cannonkeys/rekt1800/readme.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Savage65
```
Comment (awkannan) [inline keyboards/cannonkeys/rekt1800/readme.md]: TODO: Update this

**Comment by drashna** [comment]
Not sure if you've seen it, but GitHub somewhat recently added the ability to open PRs as "drafts", which may have been what you wanted here. 

No worries though,

**Comment by drashna** [comment]
Also, love the name!

**Comment by awkannan** [comment]
@drashna @noroadsleft This PR is ready now

File: keyboards/cannonkeys/rekt1800/readme.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Rekt1800
+
+Rekt1800 Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/readme.md]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```

File: keyboards/cannonkeys/rekt1800/readme.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Rekt1800
+
+Rekt1800 Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/readme.md]:
```suggestion
* Hardware Supported: STM32F072CBT6
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#MCU Name
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+#
+
+# project specific files
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#MCU Name
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+#
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#MCU Name
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/rules.mk]:
```suggestion
# MCU Name
```

File: keyboards/cannonkeys/rekt1800/info.json [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    "keyboard_name": "Rekt1800",
+    "url": "https://cannonkeys.com",
+    "maintainer": "awkannan",
+    "width": 19.5,
+    "height": 6.5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/info.json]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/cannonkeys/rekt1800/info.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
...12.75, "y":3.25}, {"label":"Enter", "x":13.75, "y":3.25, "w":1.25}, {"x":15.5, "y":3.25}, {"x":16.5, "y":3.25}, {"x":17.5, "y":3.25}, {"x":18.5, "y":3.25}, {"label":"Shift", "x":0, "y":4.25, "w":1.25}, {"x":1.25, "y":4.25}, {"label":"Z", "x":2.25, "y":4.25}, {"label":"X", "x":3.25, "y":4.25}, {"label":"C", "x":4.25, "y":4.25}, {"label":"V", "x":5.25, "y":4.25}, {"label":"B", "x":6.25, "y":4.25}, {"label":"N", "x":7.25, "y":4.25}, {"label":"M", "x":8.25, "y":4.25}, {"label":"<", "x":9.25, "y":4.25}, {"label":">", "x":10.25, "y":4.25}, {"label":"?", "x":11.25, "y":4.25}, {"label":"Shift", "x":12.25, "y":4.25, "w":1.75}, {"x":15.5, "y":4.25}, {"x":16.5, "y":4.25}, {"x":17.5, "y":4.25}, {"x":18.5, "y":4.25}, {"label":"\u2191", "x":14.25, "y":4.5}, {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, {"label":"Alt", "x":1.25, "y":5.25, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.25, "w":1.25}, {"x":3.75, "y":5.25, "w":6.25}, {"label":"Alt", "x":10, "y":5.25, "w":1.5}, {"label":"Ctrl", "x":11.5, "y":5.25, "w":1.5}, {"x":16.5, "y":5.25}, {"x":17.5, "y":5.25}, {"x":18.5, "y":5.25}, {"label":"\u2190", "x":13.25, "y":5.5}, {"label":"\u2193", "x":14.25, "y":5.5}, {"label":"\u2192", "x":15.25, "y":5.5}]
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/info.json]: Arrow keys out of sequence.

```suggestion
            "layout": [{"label":"Esc", "x":0, "y":0}, {"label":"F1", "x":1.25, "y":0}, {"label":"F2", "x":2.25, "y":0}, {"label":"F3", "x":3.25, "y":0}, {"label":"F4", "x":4.25, "y":0}, {"label":"F5", "x":5.5, "y":0}, {"label":"F6", "x":6.5, "y":0}, {"label":"F7", "x":7.5, "y":0}, {"label":"F8", "x":8.5, "y":0}, {"label":"F9", "x":9.75, "y":0}, {"label":"F10", "x":10.75, "y":0}, {"label":"F11", "x":11.75, "y":0}, {"label":"F12", "x":12.75, "y":0}, {"x":14, "y":0}, {"label":"PrtSc", "x":15.5, "y":0}, {"label":"Scroll Lock", "x":16.5, "y":0}, {"label":"Pause", "x":17.5, "y":0}, {"x":18.5, "y":0}, {"label":"~", "x":0, "y":1.25}, {"label":"!", "x":1, "y":1.25}, {"label":"@", "x":2, "y":1.25}, {"label":"#", "x":3, "y":1.25}, {"label":"$", "x":4, "y":1.25}, {"label":"%", "x":5, "y":1.25}, {"label":"^", "x":6, "y":1.25}, {"label":"&", "x":7, "y":1.25}, {"label":"*", "x":8, "y":1.25}, {"label":"(", "x":9, "y":1.25}, {"label":")", "x":10, "y":1.25}, {"label":"_", "x":11, "y":1.25}, {"label":"+", "x":12, "y":1.25}, {"x":13, "y":1.25}, {"x":14, "y":1.25}, {"label":"Insert", "x":15.5, "y":1.25}, {"label":"Home", "x":16.5, "y":1.25}, {"label":"PgUp", "x":17.5, "y":1.25}, {"x":18.5, "y":1.25}, {"label":"Tab", "x":0, "y":2.25, "w":1.5}, {"label":"Q", "x":1.5, "y":2.25}, {"label":"W", "x":2.5, "y":2.25}, {"label":"E", "x":3.5, "y":2.25}, {"label":"R", "x":4.5, "y":2.25}, {"label":"T", "x":5.5, "y":2.25}, {"label":"Y", "x":6.5, "y":2.25}, {"label":"U", "x":7.5, "y":2.25}, {"label":"I", "x":8.5, "y":2.25}, {"label":"O", "x":9.5, "y":2.25}, {"label":"P", "x":10.5, "y":2.25}, {"label":"{", "x":11.5, "y":2.25}, {"label":"}", "x":12.5, "y":2.25}, {"label":"|", "x":13.5, "y":2.25, "w":1.5}, {"label":"Delete", "x":15.5, "y":2.25}, {"label":"End", "x":16.5, "y":2.25}, {"label":"PgDn", "x":17.5, "y":2.25}, {"x":18.5, "y":2.25}, {"label":"Caps Lock", "x":0, "y":3.25, "w":1.75}, {"label":"A", "x":1.75, "y":3.25}, {"label":"S", "x":2.75, "y":3.25}, {"label":"D", "x":3.75, "y":3.25}, {"label":"F", "x":4.75, "y":3.25}, {"label":"G", "x":5.75, "y":3.25}, {"label":"H", "x":6.75, "y":3.25}, {"label":"J", "x":7.75, "y":3.25}, {"label":"K", "x":8.75, "y":3.25}, {"label":"L", "x":9.75, "y":3.25}, {"label":":", "x":10.75, "y":3.25}, {"label":"\"", "x":11.75, "y":3.25}, {"x":12.75, "y":3.25}, {"label":"Enter", "x":13.75, "y":3.25, "w":1.25}, {"x":15.5, "y":3.25}, {"x":16.5, "y":3.25}, {"x":17.5, "y":3.25}, {"x":18.5, "y":3.25}, {"label":"Shift", "x":0, "y":4.25, "w":1.25}, {"x":1.25, "y":4.25}, {"label":"Z", "x":2.25, "y":4.25}, {"label":"X", "x":3.25, "y":4.25}, {"label":"C", "x":4.25, "y":4.25}, {"label":"V", "x":5.25, "y":4.25}, {"label":"B", "x":6.25, "y":4.25}, {"label":"N", "x":7.25, "y":4.25}, {"label":"M", "x":8.25, "y":4.25}, {"label":"<", "x":9.25, "y":4.25}, {"label":">", "x":10.25, "y":4.25}, {"label":"?", "x":11.25, "y":4.25}, {"label":"Shift", "x":12.25, "y":4.25, "w":1.75}, {"label":"\u2191", "x":14.25, "y":4.5}, {"x":15.5, "y":4.25}, {"x":16.5, "y":4.25}, {"x":17.5, "y":4.25}, {"x":18.5, "y":4.25}, {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, {"label":"GUI", "x":1.25, "y":5.25, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.25, "w":1.25}, {"x":3.75, "y":5.25, "w":6.25}, {"label":"Alt", "x":10, "y":5.25, "w":1.5}, {"label":"Ctrl", "x":11.5, "y":5.25, "w":1.5}, {"label":"\u2190", "x":13.25, "y":5.5}, {"label":"\u2193", "x":14.25, "y":5.5}, {"label":"\u2192", "x":15.25, "y":5.5}, {"x":16.5, "y":5.25}, {"x":17.5, "y":5.25}, {"x":18.5, "y":5.25}]
```

File: keyboards/cannonkeys/rekt1800/rekt1800.h [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+#include "quantum.h"
+#define K_NO KC_NO
+
+#define LAYOUT_all( \
+	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012,       K014, K015, K016, K017, K018, \
+	K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, K115, K116, K117, K118, \
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212,       K214, K215, K216, K217, K218, \
+	K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312,       K314, K315, K316, K317, K318, \
+	K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412,       K414, K415, K416, K417, K418, \
+	K500, K501, K502,                   K506,                   K510, K511, K512,       K514, K515, K516, K517        \
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/rekt1800.h]: Your `info.json` file is set up for a split Numpad Enter, but there's no argument here for it. The JSON file seems to indicate that there should be a `K518` here.

https://github.com/qmk/qmk_firmware/blob/32e247653fa9e3e32fc6319314aa9324b8c01912/keyboards/cannonkeys/rekt1800/rekt1800.h#L12

```suggestion
	K500, K501, K502,                   K506,                   K510, K511, K512,       K514, K515, K516, K517, K518  \
```

Both keymaps would need to be updated for this as well.

File: keyboards/cannonkeys/rekt1800/rekt1800.h [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+) { \
+	{ K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K_NO, K014, K015, K016, K017, K018 }, \
+	{ K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, K115, K116, K117, K118 }, \
+	{ K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K_NO, K214, K215, K216, K217, K218 }, \
+	{ K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312, K_NO, K314, K315, K316, K317, K318 }, \
+    { K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412, K_NO, K414, K415, K416, K417, K418 }, \
+	{ K500, K501, K502, K_NO, K_NO, K_NO, K506, K_NO, K_NO, K_NO, K510, K511, K512, K_NO, K514, K515, K516, K517, K_NO }  \
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/rekt1800.h]:
```suggestion
	{ K500, K501, K502, K_NO, K_NO, K_NO, K506, K_NO, K_NO, K_NO, K510, K511, K512, K_NO, K514, K515, K516, K517, K518 }  \
```

File: keyboards/cannonkeys/rekt1800/rekt1800.h [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+}
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/rekt1800.h]: Also, could you clean up the indenting on this file? You're mixing tabs and spaces here. (Four-space indent is preferred per our [C Coding Conventions](https://docs.qmk.fm/#/coding_conventions_c?id=coding-conventions-c).)

File: keyboards/cannonkeys/rekt1800/rekt1800.h (hunk 11 above)
Comment (awkannan) [reply keyboards/cannonkeys/rekt1800/rekt1800.h]: There is actually no split numpad enter on the PCB. I will have to fix the JSON

File: keyboards/cannonkeys/rekt1800/info.json [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
...12.75, "y":3.25}, {"label":"Enter", "x":13.75, "y":3.25, "w":1.25}, {"x":15.5, "y":3.25}, {"x":16.5, "y":3.25}, {"x":17.5, "y":3.25}, {"x":18.5, "y":3.25}, {"label":"Shift", "x":0, "y":4.25, "w":1.25}, {"x":1.25, "y":4.25}, {"label":"Z", "x":2.25, "y":4.25}, {"label":"X", "x":3.25, "y":4.25}, {"label":"C", "x":4.25, "y":4.25}, {"label":"V", "x":5.25, "y":4.25}, {"label":"B", "x":6.25, "y":4.25}, {"label":"N", "x":7.25, "y":4.25}, {"label":"M", "x":8.25, "y":4.25}, {"label":"<", "x":9.25, "y":4.25}, {"label":">", "x":10.25, "y":4.25}, {"label":"?", "x":11.25, "y":4.25}, {"label":"Shift", "x":12.25, "y":4.25, "w":1.75}, {"label":"\u2191", "x":14.25, "y":4.5}, {"x":15.5, "y":4.25}, {"x":16.5, "y":4.25}, {"x":17.5, "y":4.25}, {"x":18.5, "y":4.25}, {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, {"label":"GUI", "x":1.25, "y":5.25, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.25, "w":1.25}, {"x":3.75, "y":5.25, "w":6.25}, {"label":"Alt", "x":10, "y":5.25, "w":1.5}, {"label":"Ctrl", "x":11.5, "y":5.25, "w":1.5}, {"label":"\u2190", "x":13.25, "y":5.5}, {"label":"\u2193", "x":14.25, "y":5.5}, {"label":"\u2192", "x":15.25, "y":5.5}, {"x":16.5, "y":5.25}, {"x":17.5, "y":5.25}, {"x":18.5, "y":5.25}]
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/info.json]: JSON fix: deletes the split Numpad Enter

```suggestion
            "layout": [{"label":"Esc", "x":0, "y":0}, {"label":"F1", "x":1.25, "y":0}, {"label":"F2", "x":2.25, "y":0}, {"label":"F3", "x":3.25, "y":0}, {"label":"F4", "x":4.25, "y":0}, {"label":"F5", "x":5.5, "y":0}, {"label":"F6", "x":6.5, "y":0}, {"label":"F7", "x":7.5, "y":0}, {"label":"F8", "x":8.5, "y":0}, {"label":"F9", "x":9.75, "y":0}, {"label":"F10", "x":10.75, "y":0}, {"label":"F11", "x":11.75, "y":0}, {"label":"F12", "x":12.75, "y":0}, {"x":14, "y":0}, {"label":"PrtSc", "x":15.5, "y":0}, {"label":"Scroll Lock", "x":16.5, "y":0}, {"label":"Pause", "x":17.5, "y":0}, {"x":18.5, "y":0}, {"label":"~", "x":0, "y":1.25}, {"label":"!", "x":1, "y":1.25}, {"label":"@", "x":2, "y":1.25}, {"label":"#", "x":3, "y":1.25}, {"label":"$", "x":4, "y":1.25}, {"label":"%", "x":5, "y":1.25}, {"label":"^", "x":6, "y":1.25}, {"label":"&", "x":7, "y":1.25}, {"label":"*", "x":8, "y":1.25}, {"label":"(", "x":9, "y":1.25}, {"label":")", "x":10, "y":1.25}, {"label":"_", "x":11, "y":1.25}, {"label":"+", "x":12, "y":1.25}, {"x":13, "y":1.25}, {"x":14, "y":1.25}, {"label":"Insert", "x":15.5, "y":1.25}, {"label":"Home", "x":16.5, "y":1.25}, {"label":"PgUp", "x":17.5, "y":1.25}, {"x":18.5, "y":1.25}, {"label":"Tab", "x":0, "y":2.25, "w":1.5}, {"label":"Q", "x":1.5, "y":2.25}, {"label":"W", "x":2.5, "y":2.25}, {"label":"E", "x":3.5, "y":2.25}, {"label":"R", "x":4.5, "y":2.25}, {"label":"T", "x":5.5, "y":2.25}, {"label":"Y", "x":6.5, "y":2.25}, {"label":"U", "x":7.5, "y":2.25}, {"label":"I", "x":8.5, "y":2.25}, {"label":"O", "x":9.5, "y":2.25}, {"label":"P", "x":10.5, "y":2.25}, {"label":"{", "x":11.5, "y":2.25}, {"label":"}", "x":12.5, "y":2.25}, {"label":"|", "x":13.5, "y":2.25, "w":1.5}, {"label":"Delete", "x":15.5, "y":2.25}, {"label":"End", "x":16.5, "y":2.25}, {"label":"PgDn", "x":17.5, "y":2.25}, {"x":18.5, "y":2.25}, {"label":"Caps Lock", "x":0, "y":3.25, "w":1.75}, {"label":"A", "x":1.75, "y":3.25}, {"label":"S", "x":2.75, "y":3.25}, {"label":"D", "x":3.75, "y":3.25}, {"label":"F", "x":4.75, "y":3.25}, {"label":"G", "x":5.75, "y":3.25}, {"label":"H", "x":6.75, "y":3.25}, {"label":"J", "x":7.75, "y":3.25}, {"label":"K", "x":8.75, "y":3.25}, {"label":"L", "x":9.75, "y":3.25}, {"label":":", "x":10.75, "y":3.25}, {"label":"\"", "x":11.75, "y":3.25}, {"x":12.75, "y":3.25}, {"label":"Enter", "x":13.75, "y":3.25, "w":1.25}, {"x":15.5, "y":3.25}, {"x":16.5, "y":3.25}, {"x":17.5, "y":3.25}, {"x":18.5, "y":3.25}, {"label":"Shift", "x":0, "y":4.25, "w":1.25}, {"x":1.25, "y":4.25}, {"label":"Z", "x":2.25, "y":4.25}, {"label":"X", "x":3.25, "y":4.25}, {"label":"C", "x":4.25, "y":4.25}, {"label":"V", "x":5.25, "y":4.25}, {"label":"B", "x":6.25, "y":4.25}, {"label":"N", "x":7.25, "y":4.25}, {"label":"M", "x":8.25, "y":4.25}, {"label":"<", "x":9.25, "y":4.25}, {"label":">", "x":10.25, "y":4.25}, {"label":"?", "x":11.25, "y":4.25}, {"label":"Shift", "x":12.25, "y":4.25, "w":1.75}, {"label":"\u2191", "x":14.25, "y":4.5}, {"x":15.5, "y":4.25}, {"x":16.5, "y":4.25}, {"x":17.5, "y":4.25}, {"x":18.5, "y":4.25, "h":2}, {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, {"label":"GUI", "x":1.25, "y":5.25, "w":1.25}, {"label":"Alt", "x":2.5, "y":5.25, "w":1.25}, {"x":3.75, "y":5.25, "w":6.25}, {"label":"Alt", "x":10, "y":5.25, "w":1.5}, {"label":"Ctrl", "x":11.5, "y":5.25, "w":1.5}, {"label":"\u2190", "x":13.25, "y":5.5}, {"label":"\u2193", "x":14.25, "y":5.5}, {"label":"\u2192", "x":15.25, "y":5.5}, {"x":16.5, "y":5.25}, {"x":17.5, "y":5.25}]
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/rules.mk]: Aligns the comments. No logic changes.

```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover
CUSTOM_MATRIX = no          # Custom matrix file
```

File: keyboards/cannonkeys/rekt1800/rekt1800.h (hunk 13 above)
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/rekt1800.h]: Aligns the indenting to four spaces.

```suggestion
#define LAYOUT_all( \
    K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012,       K014, K015, K016, K017, K018, \
    K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, K115, K116, K117, K118, \
    K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212,       K214, K215, K216, K217, K218, \
    K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312,       K314, K315, K316, K317, K318, \
    K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412,       K414, K415, K416, K417, K418, \
    K500, K501, K502,                   K506,                   K510, K511, K512,       K514, K515, K516, K517        \
) { \
    { K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K_NO, K014, K015, K016, K017, K018 }, \
    { K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, K115, K116, K117, K118 }, \
    { K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K_NO, K214, K215, K216, K217, K218 }, \
    { K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K312, K_NO, K314, K315, K316, K317, K318 }, \
    { K400, K401, K402, K403, K404, K405, K406, K407, K408, K409, K410, K411, K412, K_NO, K414, K415, K416, K417, K418 }, \
    { K500, K501, K502, K_NO, K_NO, K_NO, K506, K_NO, K_NO, K_NO, K510, K511, K512, K_NO, K514, K515, K516, K517, K_NO }  \
}
```

File: keyboards/cannonkeys/rekt1800/bootloader_defs.h [hunk 16]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/bootloader_defs.h]: This file can be deleted, so as to use `drivers/boards/GENERIC_STM32_F072XB/bootloader_defs.h`.

File: keyboards/cannonkeys/rekt1800/config.h [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x2377
+#define DEVICE_VER      0x0001
+/* in python2: list(u"whatever".encode('utf-16-le')) */
+/*   at most 32 characters or the ugly hack in usb_main.c borks */
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/config.h]:
```suggestion
```

**Comment by awkannan** [comment]
@fauxpark @noroadsleft @zvecr Brought this one up to date again, ready for another review

File: keyboards/cannonkeys/rekt1800/readme.md [hunk 18]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys Rekt1800
+
+Rekt1800 Keyboard
```
Comment (noroadsleft) [inline keyboards/cannonkeys/rekt1800/readme.md:3]: This should link to a photo of the board (renders are okay), or be deleted entirely.

Comment (awkannan) [reply keyboards/cannonkeys/rekt1800/readme.md:3]: Is this documented anywhere? For future reference

Comment (noroadsleft) [reply keyboards/cannonkeys/rekt1800/readme.md:3]: The preferred way of starting a new keyboard project for QMK is to run the [`util/new_keyboard.sh`](https://github.com/qmk/qmk_firmware/blob/4b10235f6777db3e7a7feab301542a0e78083b20/util/new_keyboard.sh) script, which generates all the files needed to program a keyboard to run QMK.

Both the [AVR](https://github.com/qmk/qmk_firmware/blob/4b10235f6777db3e7a7feab301542a0e78083b20/quantum/template/avr/readme.md) and [PS2AVRGB](https://github.com/qmk/qmk_firmware/blob/4b10235f6777db3e7a7feab301542a0e78083b20/quantum/template/ps2avrgb/readme.md) templates include a line for a photo on Line 3.

That script doesn't have any support for Arm microcontrollers, so I don't really blame you or other Arm-powered keyboard contributors for not running the script (and I would personally debate how many people even know about it, just based on the PRs we get for new keyboards). That script will hopefully soon be replaced by a Python script to meet the same need, and the Python script will add support for STM32 controllers, however.

**Comment by awkannan** [comment]
@noroadsleft Done, please review!
@tzarc @zvecr @fauxpark This is ready for review

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 19]
```diff
@@ -0,0 +1,19 @@
+# MCU Name
```
Comment (fauxpark) [inline keyboards/cannonkeys/rekt1800/rules.mk:1]:
```suggestion
# MCU name
```

File: keyboards/cannonkeys/rekt1800/rules.mk [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/cannonkeys/rekt1800/rules.mk:12]:
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**